src/main/java/io/confluent/connect/elasticsearch/ElasticsearchSinkConnectorConfig.java

```diff
@@ -328,11 +301,57 @@
       + " authentication with Kerberos.";
   private static final String KERBEROS_KEYTAB_PATH_DEFAULT = null;
 
+  // Data stream configs
+  public static final String DATA_STREAM_DATASET_CONFIG = "data.stream.dataset";
+  private static final String DATA_STREAM_DATASET_DOC =
+      "Generic name describing data ingested and its structure to be written to a data stream. Can"
+          + " be any arbitrary string that is no longer than 100 characters, is in all lowercase, "
+          + "and does not contain spaces or any of these special characters ``/\\*\"<>|,#:-``. "
+          + "Otherwise, no value indicates the connector will write to regular indices instead. "
+          + "If set, this configuration will be used alongside ``data.stream.type`` to "
+          + "construct the data stream name in the form of {``data.stream.type``"
+          + "}-{" + DATA_STREAM_DATASET_CONFIG + "}-{topic}.";
+  private static final String DATA_STREAM_DATASET_DISPLAY = "Data Stream Dataset";
+  private static final String DATA_STREAM_DATASET_DEFAULT = "";
+
+  public static final String DATA_STREAM_TYPE_CONFIG = "data.stream.type";
+  private static final String DATA_STREAM_TYPE_DOC = String.format(
+      "Generic type describing the data to be written to data stream. "
+          + "The default is %s which indicates the connector will write "
+          + "to regular indices instead. If set, this configuration will "
+          + "be used alongside %s to construct the data stream name in the form of"
+          + "{%s}-{%s}-{topic}.",
+      DataStreamType.NONE.name(),
+      DATA_STREAM_DATASET_CONFIG,
+      DATA_STREAM_TYPE_CONFIG,
+      DATA_STREAM_DATASET_CONFIG
+  );
+  private static final String DATA_STREAM_TYPE_DISPLAY = "Data Stream Type";
+  private static final DataStreamType DATA_STREAM_TYPE_DEFAULT = DataStreamType.NONE;
+
+  public static final String DATA_STREAM_TIMESTAMP_CONFIG = "data.stream.timestamp.field";
+  private static final String DATA_STREAM_TIMESTAMP_DOC = String.format(
+      "The Kafka record field to use as the timestamp for the ``@timestamp`` field in documents "
+          + "sent to a data stream.\n All documents sent to a data stream needs an``@timestamp`` "
+          + " field with values of type ``date`` or ``data_nanos``. Otherwise, the document "
+          + " would not be sent. If multiple fields are provided, the first field listed that "
+          + "also appears in the record will be used. If this configuration is left empty,"
+          + " all of the documents will use the record timestamp as the ``@timestamp`` field "
+          + "value. Note that the ``@timestamp`` field needs to be explicitly listed if records "
+          + "already contain this field. This configuration can only be set if ``%s`` and ``%s`` "
+          + "are set.",
```

Comment:
```suggestion
  private static final String DATA_STREAM_TIMESTAMP_DOC = String.format(
      "The Kafka record field to use as the timestamp for the ``@timestamp`` field in documents "
          + "sent to a data stream.\n All documents sent to a data stream needs an``@timestamp`` "
          + " field with values of type ``date`` or ``data_nanos``. Otherwise, the document "
          + " will not be sent. If multiple fields are provided, the first field listed that "
          + "also appears in the record will be used. If this configuration is left empty,"
          + " all of the documents will use the Kafka record timestamp as the ``@timestamp`` field "
          + "value. Note that ``@timestamp``still  needs to be explicitly listed if records "
          + "already contain this field. This configuration can only be set if ``%s`` and ``%s`` "
          + "are set.",
```